code/game/objects/items/sticker.dm

```diff
@@ -0,0 +1,116 @@
+/obj/item/sticker
+	name = "sticker"
+	desc = "A sticker with some strong adhesive on the back, sticks to stuff!"
+	item_flags = NOBLUDGEON | XENOMORPH_HOLDABLE //funny
+	resistance_flags = FLAMMABLE
+	icon = 'icons/obj/stickers.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	throw_range = 3
+	vis_flags = VIS_INHERIT_DIR | VIS_INHERIT_PLANE | VIS_INHERIT_LAYER
+	var/mutable_appearance/sticker_overlay
+	var/list/icon_states
+	var/atom/attached
+
+/obj/item/sticker/Initialize(mapload)
+	. = ..()
+	if(icon_states)
+		icon_state = pick(icon_states)
+	pixel_y = rand(-3,3)
+	pixel_x = rand(-3,3)
+
+/obj/item/sticker/afterattack(atom/target, mob/living/user, prox, params)
+	. = ..()
+	if(!prox)
+		return
+	if(!isliving(target) && !isobj(target) && !isturf(target))
+		return
+	user.visible_message(span_notice("[user] sticks [src] to [target]!"),span_notice("You stick [src] to [target]!"))
+	var/list/parameters = params2list(params)
+	var/divided_size = world.icon_size / 2
+	var/py = text2num(parameters["icon-y"]) - divided_size
+	var/px = text2num(parameters["icon-x"]) - divided_size
+	. |= AFTERATTACK_PROCESSED_ITEM
+	stick(target,user,px,py)
+
+/obj/item/sticker/proc/stick(atom/target, mob/living/user, px,py)
+	sticker_overlay = mutable_appearance(icon, icon_state , layer = target.layer + 1, appearance_flags = RESET_COLOR | PIXEL_SCALE)
+	sticker_overlay.pixel_x = px
+	sticker_overlay.pixel_y = py
+	target.add_overlay(sticker_overlay)
+	attached = target
+	register_signals(user)
+	moveToNullspace()
+
+/obj/item/sticker/proc/peel(datum/source, silent=FALSE)
+	SIGNAL_HANDLER
+	if(!attached)
+		return
+	attached.cut_overlay(sticker_overlay)
+	sticker_overlay = null
+	forceMove(isturf(attached) ? attached : attached.loc)
+	if(!silent)
+		attached.visible_message(span_notice("[src] falls off [attached]."))
+	pixel_y = rand(-3,3)
+	pixel_x = rand(-3,3)
+	unregister_signals()
+	attached = null
+
+/obj/item/sticker/proc/register_signals(mob/living/user)
+	RegisterSignal(attached, COMSIG_COMPONENT_CLEAN_ACT, PROC_REF(peel))
+	if(isturf(attached))
+		//register signals on the users turf instead because we can assume they are on flooring sticking it to a wall so it should burn (otherwise it would fruitlessly check wall temperature)
+		RegisterSignal((user && isclosedturf(attached)) ? get_turf(user) : attached, COMSIG_TURF_EXPOSE, PROC_REF(on_turf_expose))
+	RegisterSignal(attached, COMSIG_LIVING_IGNITED, PROC_REF(on_ignite))
+	RegisterSignal(attached, COMSIG_PARENT_QDELETING, PROC_REF(unregister_signals))
```

Comment:
Maybe I am missing something here, but is this signal not checking that the thing it is attached to has been deleted?
Once it has been deleted it is not going to be sending any signals about being cleaned by space cleaner because it doesn't exist any more, leaving the sticker in nullspace forever.

Reply:
You also don't need to unregister signals on a deleted object, it handles that itself.

Reply:
Now you're not unregistering the signals on deletion but is it not still true that if something you are attached to is destroyed and it is _not_ via fire or temperature, your item will sit forever in nullspace?
I think you needed the `COMSIG_PARENT_QDELETING` signal but should simply delete yourself upon receiving it.